expression/aggregation/descriptor.go

```diff
@@ -37,6 +37,8 @@ type AggFuncDesc struct {
 	HasDistinct bool
 	// OrderByItems represents the order by clause used in GROUP_CONCAT
 	OrderByItems []*util.ByItems
+	// IsWindowAggFunc represents the aggregation function is used to window function.
+	IsWindowAggFunc bool
```

Comment:
Could we avoid involving `IsWindowAggFunc`? 
The transformation from `maxMin4XXX` to `maxMin4XXXSliding` can be done in [`BuildWindowFunctions`](https://github.com/pingcap/tidb/blob/9047a9c8814b40880592980defd1988d9a222fb6/executor/aggfuncs/builder.go#L67) simply.

Reply:
get!